rust/hyperlane-base/src/contract_sync/cursors/mod.rs

```diff
@@ -13,8 +13,8 @@ pub enum CursorType {
     RateLimited,
 }
 
-// H512 * 1M = 64MB per origin chain
-const TX_ID_CHANNEL_CAPACITY: Option<usize> = Some(1_000_000);
+// H512 * 30k =~ 2MB per origin chain
+const TX_ID_CHANNEL_CAPACITY: Option<usize> = Some(30_000);
```

Comment:
Setting this to 30k since that's the max we've seen empirically. This will get rounded up to `32,768` - overall a 97% reduction in memory usage